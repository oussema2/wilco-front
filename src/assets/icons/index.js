/* eslint-disable global-require */
export const aircraft = require( './aircraft.png' );
export const airport = require( './airport.png' );
export const arrowLeft = require( './arrowLeft.png' );
export const arrowRight = require( './arrowRight.png' );
export const award = require( './award.png' );
export const ban = require( './ban.png' );
export const bell = require( './bell.png' );
export const binoculars = require( './binoculars.png' );
export const brandSymbol = require( './brandSymbol.png' );
export const chevronLeft = require( './chevronLeft.png' );
export const chevronRight = require( './chevronRight.png' );
export const cloud = require( './cloud.png' );
export const certificate = require( './certificate.png' );
export const chatComment = require( './chatComment.png' );
export const chatNewMessage = require( './chatNewMessage.png' );
export const checkbox = require( './checkbox.png' );
export const checkboxUnselected = require( './checkboxUnselected.png' );
export const comment = require( './comment.png' );
export const cross = require( './cross.png' );
export const eye = require( './eye.png' );
export const eyeSlash = require( './eyeSlash.png' );
export const exclamationCircle = require( './exclamationCircle.png' );
export const exclamationTriangle = require( './exclamationTriangle.png' );
export const filter = require( './filter.png' );
export const flightInfoAircraft = require( './flightInfoAircraft.png' );
export const home = require( './home.png' );
export const horizontalLogo = require( './horizontalLogo.png' );
export const infoCircle = require( './infoCircle.png' );
export const logOut = require( './logOut.png' );
export const map = require( './map.png' );
export const meatballsMenu = require( './meatballsMenu.png' );
export const megaphone = require( './megaphone.png' );
export const onBoardingFirst = require( './onBoardingFirst.png' );
export const onBoardingSecond = require( './onBoardingSecond.png' );
export const onBoardingThird = require( './onBoardingThird.png' );
export const planeDeparture = require( './planeDeparture.png' );
export const plus = require( './plus.png' );
export const photoVideo = require( './photoVideo.png' );
export const privacyOnlyMe = require( './privacyOnlyMe.png' );
export const privacyPublic = require( './privacyPublic.png' );
export const remove = require( './remove.png' );
export const route = require( './route.png' );
export const rulerVertical = require( './rulerVertical.png' );
export const star = require( './star.png' );
export const starMyFeed = require( './starMyFeed.png' );
export const search = require( './search.png' );
export const settings = require( './settings.png' );
export const sliders = require( './sliders.png' );
export const speed = require( './speed.png' );
export const stopwatch = require( './stopwatch.png' );
export const thumbsUp = require( './thumbsUp.png' );
export const user = require( './user.png' );
export const users = require( './users.png' );
export const vector = require( './vector.png' );
