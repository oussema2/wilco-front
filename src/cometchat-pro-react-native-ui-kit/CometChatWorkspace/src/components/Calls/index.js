export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
