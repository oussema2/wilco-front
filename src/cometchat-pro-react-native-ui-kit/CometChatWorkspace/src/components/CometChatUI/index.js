/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
