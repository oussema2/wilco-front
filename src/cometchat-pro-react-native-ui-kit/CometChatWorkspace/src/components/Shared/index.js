export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
