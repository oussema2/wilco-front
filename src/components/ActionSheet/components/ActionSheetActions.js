import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
