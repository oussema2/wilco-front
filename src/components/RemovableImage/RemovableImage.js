import React from 'react';
import PropTypes from 'prop-types';
import {
	ImageBackground, ViewPropTypes
} from 'react-native';
import RemoveImageButton from './components/RemoveImageButton';
import { imageStyles } from './RemovableImage.styles';

const RemovableImage = ( {
	source, testID, size, tintColor, style, resizeMode, removeOnPress
} ) => (
	<ImageBackground
		testID={testID}
		style={{ ...imageStyles( { size, tintColor } ).image, ...style }}
		resizeMode={resizeMode}
		source={source}
	>
		<RemoveImageButton onPress={removeOnPress} />
	</ImageBackground>
);

RemovableImage.propTypes = {
	source: PropTypes.oneOfType( [
		PropTypes.instanceOf( Object ),
		PropTypes.number
	] ).isRequired,
	testID: PropTypes.string,
	size: PropTypes.number,
	tintColor: PropTypes.string,
	style: ViewPropTypes.style,
	resizeMode: PropTypes.string,
	removeOnPress: PropTypes.func.isRequired
};

RemovableImage.defaultProps = {
	testID: 'image',
	size: 32,
	tintColor: null,
	style: {},
	resizeMode: 'cover'
};

export default RemovableImage;
