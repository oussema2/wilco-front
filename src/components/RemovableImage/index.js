export { default as RemovableImage } from './RemovableImage';
