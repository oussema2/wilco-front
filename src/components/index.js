export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
