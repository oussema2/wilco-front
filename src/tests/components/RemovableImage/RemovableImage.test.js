import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { RemovableImage } from '../../../components/RemovableImage';
import { home } from '../../../assets/icons';

describe( 'RemovableImage', () => {
	const defaultImage = home;
	const removeOnPress = jest.fn();

	describe( 'with the default props', () => {
		it( 'renders the RemovableImage component correctly', async () => {
			const { component, queryByTestId } = render(
				<RemovableImage source={defaultImage} removeOnPress={removeOnPress} />
			);
			expect( queryByTestId( 'image' ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when the remove button is pressed', () => {
		it( 'calls removeOnPress', () => {
			const { queryByTestId } = render(
				<RemovableImage source={defaultImage} removeOnPress={removeOnPress} />
			);
			fireEvent.press( queryByTestId( 'removeImageButton' ) );
			expect( removeOnPress ).toHaveBeenCalledTimes( 1 );
		} );
	} );
} );
