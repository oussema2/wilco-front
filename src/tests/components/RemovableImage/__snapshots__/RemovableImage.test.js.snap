// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemovableImage with the default props renders the RemovableImage component correctly 1`] = `undefined`;
